src/phases/query/CollScanPredicateSelectivity.yml

```diff
@@ -0,0 +1,427 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of conjunctive collection scan queries where the order of
+  predicates matters due to selectivity of the predicates.
+
+GlobalDefaults:
+  Database: &Database {^Parameter: {Name: Database, Default: unused}}
+  DocumentCount: &DocumentCount {^Parameter: {Name: DocumentCount, Default: -1}}
+  Repeat: &Repeat {^Parameter: {Name: Repeat, Default: -1}}
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 31
+  Billion: &Billion 1_000_000_000
+  NegativeBillion: &NegativeBillion -1_000_000_000
+
+ActorTemplates:
+- TemplateName: FindQueryTemplate
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "unused"}}
+    Type: CrudActor
+    Database: *Database
+    Threads: 1
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "ActivePhase", Default: -1}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Repeat: *Repeat
+          Collection: *Collection
+          Operations:
+          - OperationName: find
+            OperationCommand:
+              Filter: {^Parameter: {Name: "Filter", Default: {invalid: "invalid"}}}
```

Comment:
What happens if the default filter is ever used? I looks like it will cause an unindexed collection scan ... the results may not be different enough from a low selectivity scan to be caught.

I'm not sure how to deal with this ... it may be a genny feature request (e.g. make a parameter mandatory and fail  if there isn't one) or double checking if this is the current behaviour. I'm pretty sure its not.

Reply:
Could just set the filter to the value null by default, which would crash the query?

Reply:
We could use null assuming that it gives a reasonable error in the task. 

Note: I'm not proposing holding up the review.... If it's a feature request we can just add a TODO to come back to.

Reply:
That works for me, I've also raised DEVPROD-3649 to request mandatory parameter support.

Reply:
That works for me, I've also raised DEVPROD-3649 to request mandatory parameter support.